tests/functional/idempotency/test_idempotency.py

```diff
@@ -690,3 +691,33 @@ def test_raise_on_no_idempotency_key(persistence_store):
 
     # THEN raise IdempotencyKeyError error
     assert "No data found to create a hashed idempotency_key" in str(excinfo.value)
+
+
+@pytest.mark.parametrize("persistence_store", [{"use_local_cache": True}], indirect=True)
+def test_jmespath_with_powertools_json(persistence_store):
+    # GIVEN an event_key_jmespath with powertools_json custom function
+    persistence_store.event_key_jmespath = "[requestContext.authorizer.claims.sub, powertools_json(body).id]"
+    persistence_store.event_key_compiled_jmespath = jmespath.compile(persistence_store.event_key_jmespath)
```

Comment:
do you actually need this? shouldn't this be covered here:

```python
        if self.event_key_jmespath:
            self.event_key_compiled_jmespath = jmespath.compile(event_key_jmespath)
```

This key `event_key_compiled_jmespath` sounds like it's primarily cache -- JMESPath search function already caches it after the first result, so calling it `compile` doesn't gain much tbh (feel free to do a perf test if you'd like)

https://github.com/jmespath/jmespath.py/blob/develop/jmespath/parser.py#L85